moveit_planners/ompl/ompl_interface/test/test_state_validity_checker.cpp

```diff
@@ -0,0 +1,350 @@
+    ompl_state->as<ompl_interface::JointModelStateSpace::StateType>()->clearKnownInformation();
+
+    ROS_DEBUG_STREAM_NAMED(LOGNAME, ompl_state.reals());
+
+    bool result_2 = checker->isValid(ompl_state.get());
+    EXPECT_FALSE(result_2);
+  }
+
+  /** This test takes a state that is known to be in self-collision and inside the joint limits as input. **/
+  void testSelfCollision(const std::vector<double>& position_in_self_collision)
+  {
+    // create a validity checker for this test
+    auto checker = std::make_shared<ompl_interface::StateValidityChecker>(planning_context_.get());
+    checker->setVerbose(VERBOSE);
+
+    robot_state_->setJointGroupPositions(joint_model_group_, position_in_self_collision);
+
+    // use a scoped OMPL state so we don't have to call allocState and freeState
+    // (as recommended in the OMPL documantion)
+    ompl::base::ScopedState<> ompl_state(state_space_);
+    state_space_->copyToOMPLState(ompl_state.get(), *robot_state_);
+
+    ROS_DEBUG_STREAM_NAMED(LOGNAME, ompl_state.reals());
+
+    // the given state is known to be in self-collision, we check it here
+    bool result = checker->isValid(ompl_state.get());
+    EXPECT_FALSE(result);
+
+    // but it should respect the joint limits
+    bool result_2 = robot_state_->satisfiesBounds();
+    EXPECT_TRUE(result_2);
+  }
+
+  void testPathConstraints(const std::vector<double>& position_in_joint_limits)
+  {
+    ASSERT_NE(planning_context_, nullptr) << "Initialize planning context before adding path constraints.";
+
+    // set the robot to a known position that is withing the joint limits and collision free
+    robot_state_->setJointGroupPositions(joint_model_group_, position_in_joint_limits);
+
+    // create position constraints around the given robot state
+    moveit_msgs::Constraints path_constraints;
+    Eigen::Isometry3d ee_pose = robot_state_->getGlobalLinkTransform(ee_link_name_);
+    path_constraints.name = "test_position_constraints";
+    path_constraints.position_constraints.push_back(createPositionConstraint(
+        { ee_pose.translation().x(), ee_pose.translation().y(), ee_pose.translation().z() }, { 0.1, 0.1, 0.1 }));
+
+    moveit_msgs::MoveItErrorCodes error_code_not_used;
+    bool succeeded = planning_context_->setPathConstraints(path_constraints, &error_code_not_used);
+    ASSERT_TRUE(succeeded) << "Failed to set path constraints on the planning context";
+
+    auto checker = std::make_shared<ompl_interface::StateValidityChecker>(planning_context_.get());
+    checker->setVerbose(VERBOSE);
+
+    // use a scoped OMPL state so we don't have to call allocState and freeState
+    // (as recommended in the OMPL documantion)
+    ompl::base::ScopedState<> ompl_state(state_space_);
+    state_space_->copyToOMPLState(ompl_state.get(), *robot_state_);
+
+    ROS_DEBUG_STREAM_NAMED(LOGNAME, ompl_state.reals());
+
+    bool result = checker->isValid(ompl_state.get());
+    EXPECT_TRUE(result);
```

Comment:
```suggestion
    EXPECT_TRUE(checker->isValid(ompl_state.get()));
```

That should produce nicer output when it fails.  Otherwise you are just going to get something that complains about the statement "result" being false.  The other thing you could do is use the stream operator like you did with the ASSERT statement on 171 to provide a nicer error message.

Reply:
Good tip, thanks! I will fix it tomorrow for all the instances of this thing.